src/index.ts

```diff
@@ -26,13 +26,19 @@ function loader(this: interfaces.Webpack, contents: string) {
         return;
     }
 
+    // Load files generated by previous loaders in the same chain
+    const generatedFiles = getGeneratedFiles(this);
+    for (let filePath of Object.keys(generatedFiles)) {
+        updateFileInCache(filePath, generatedFiles[filePath], instance);
+    }
+
     const rawFilePath = path.normalize(this.resourcePath);
     const filePath = utils.appendTsSuffixIfMatch(options.appendTsSuffixTo, rawFilePath);
     const fileVersion = updateFileInCache(filePath, contents, instance);
 
     const { outputText, sourceMapText } = options.transpileOnly
         ? getTranspilationEmit(filePath, contents, instance, this)
```

Comment:
Hmm... I see that the above may not be entirely clear. The reason `getEmit` recieves `generatedFiles` is that it add dependencies, and they need to be filtered so that generated files are excluded. `getTranspilationEmit` does not add dependencies and so does not need `generatedFiles`.